lib/app.js

```diff
@@ -35,47 +40,112 @@ const {
 const INITIAL_SCAN_PERIOD = 8000; // 8secs
 
 /**
- * Main app view
+ * Main app view.
+ *
+ * Has no visual state, acts as 'container' for
+ * app UI, composing components and supplying
+ * them with the correct data.
+ *
+ * Uses `Navigator` to split the views into different 'scenes'.
+ *
+ * The parent `Provider` view passes the ReduxStore
+ * `store`. We use 'react-redux' to expose *parts*
+ * of this store to this view's `this.props.*`.
  *
  * @type {ReactComponent}
  */
-class AppView extends React.Component {
+class App extends React.Component {
   constructor(props) {
     super(props);
 
     track.appLaunch();
     this.scanning = false;
-    this.scanner = props.scanner || new Scanner({  // eslint-disable-line
-      onUpdate: actions.updateItem,
-      onLost: actions.removeItem,
-      shouldPopulateItem(url) {
-        return !store.getState().items
-          .some(item => item.originalUrl === url);
-      }
+
+    // create the scanner
+    this.scanner = new Scanner({
+      onUpdate: this.props.updateItem,
+      onLost: this.props.removeItem,
+      onMetadata: this.props.setItemMetadata,
+      expiryEnabled: this.enabled.bind(this, 'removeOldItems'),
+      getItems: () => this.props.items
     });
 
     this.setupNotificationListeners();
+    this.inflateUserFlags();
```

Comment:
Not manipulated, but loaded in memory, even if we don't access settings.

Again, nothing to worry right now, but something to think in the future.